source/common/quic/envoy_quic_client_stream.cc

```diff
@@ -80,30 +80,38 @@ void EnvoyQuicClientStream::encodeData(Buffer::Instance& data, bool end_stream)
   ASSERT(!local_end_stream_);
   local_end_stream_ = end_stream;
   SendBufferMonitor::ScopedWatermarkBufferUpdater updater(this, this);
-  Buffer::RawSliceVector raw_slices = data.getRawSlices();
-  absl::InlinedVector<quiche::QuicheMemSlice, 4> quic_slices;
-  quic_slices.reserve(raw_slices.size());
-  for (auto& slice : raw_slices) {
-    ASSERT(slice.len_ != 0);
-    // Move each slice into a stand-alone buffer.
-    // TODO(danzh): investigate the cost of allocating one buffer per slice.
-    // If it turns out to be expensive, add a new function to free data in the middle in buffer
-    // interface and re-design QuicheMemSliceImpl.
-    quic_slices.emplace_back(quiche::QuicheMemSlice::InPlace(), data, slice.len_);
-  }
-  quic::QuicConsumedData result{0, false};
-  absl::Span<quiche::QuicheMemSlice> span(quic_slices);
-  {
+  if (http_datagram_handler_) {
     IncrementalBytesSentTracker tracker(*this, *mutableBytesMeter(), false);
```

Comment:
It looks like we're no longer using this tracker in the non-datagram case. Is that right?

Reply:
Nothing has changed for the non-datagram case. I put another tracker object here to count the bytes sent for the datagram encoding case, but I wasn't sure this is necessary when I put it here. Can I just remove the tracker for the datagram case?